plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -603,7 +603,7 @@ public function update_rest_query_in_editor( $args, $request ): array {
 		}
 
 		$orderby             = $request->get_param( 'orderBy' );
-		$on_sale             = $request->get_param( 'woocommerceOnSale' ) === 'true';
+		$on_sale             = is_null( $request->get_param( 'woocommerceOnSale' ) ) ? null : $request->get_param( 'woocommerceOnSale' );
```

Comment:
Nitpick: Maybe we can use `??` to simplify the code?

```suggestion
		$on_sale             = $request->get_param( 'woocommerceOnSale' ) ?? null;
```

Reply:
No need to manage this here.
Done -> [5591462](https://github.com/woocommerce/woocommerce/pull/50808/commits/55914625860a73462fb8f63e59177770178ae3f7)